lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -493,36 +492,27 @@ def archive(self, trans, history_id, filename='', format='zip', dry_run=True, **
         return self.service.archive(trans, history_id, filter_parameters, filename, dry_run)
 
     @expose_api_raw_anonymous
-    def contents_near(self, trans, history_id, hid, limit, **kwd):
+    def contents_near(self, trans, history_id, direction, hid, limit, **kwd):
         """
-        Return {limit} history items "near" {hid}, so that |before| <= limit // 2, |after| <= limit // 2 + 1.
-        Additional counts provided in the HTTP headers.
+        Returns the following data:
 
-        GET /api/histories/{history_id}/contents/near/{hid}/{limit}
-        """
-        return self._handle_direction(trans, history_id, DirectionOptions.near, hid, limit, **kwd)
-
-    @expose_api_raw_anonymous
-    def contents_after(self, trans, history_id, hid, limit, **kwd):
-        """
-        Return {limit} history items with hid > {hid}.
-        Additional counts provided in the HTTP headers.
+        a) item counts:
+        - total matches-up:   hid < {hid}
+        - total matches-down: hid > {hid}
+        - total matches:      total matches-up + total matches-down + 1 (+1 for hid == {hid})
```

Comment:
It looks like this is computed as just total-matches-up + total-matches-down here: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/services/history_contents.py#:~:text=counts%5B%27total_matches%27%5D%20%3D%20total_matches_up%20%2B%20total_matches_down

Do we need to add one there to get the total including the given hid?

Reply:
yes, that was incorrect. Fixed. Thanks!

Do you think the counts for displayed matches should follow the same pattern - i.e., displayed-matches-up and displayed-matches-down would NOT include the item where hid == {hid}, but the total displayed matches would? (It's trivial to adjust if needed, unless that would affect the current virtual scroller)

Reply:
I think it would make more sense to just say `total matches` in the response instead of matches-up/matches-down/matches, since now the up/down/near is explicitly requested?

Reply:
As I understand, each of theses counts are needed for the new scroller @assuntad23  is working on: total matches (above hid/below hid/all) which are counts of items that match the filters excluding the up/down/near direction, as well as displayed items (that are displayed above/below his + all displayed). That's where the 6 counts came from.

Reply:
@ic4f Thought about this more and while I think we can likely simplify this as long as we follow up within the release we can iterate on this and I don't want to hold up the merge here.  Thanks for this!

Reply:
@dannon I completely agree - I also feel this can be simplified, so yes, let's revisit this as the scroller takes shape. Thanks for merging!